sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsAsyncClient.java

```diff
@@ -0,0 +1,1371 @@
+                .setCountryHint(detectLanguageInput.getCountryHint())).collect(Collectors.toList()));
+
+        return service.languagesWithRestResponseAsync(
+            languageBatchInput, options == null ? null : options.getModelVersion(),
+            options == null ? null : options.showStatistics(), context)
+            .doOnSubscribe(ignoredValue -> logger.info("A batch of language input - {}", ignoredValue))
+            .doOnSuccess(response -> logger.info("A batch of detected language output - {}", response.getValue()))
+            .doOnError(error -> logger.warning("Failed to detected languages - {}", error))
+            .map(response -> new SimpleResponse<>(response, toDocumentResultCollection(response.getValue())));
+    }
+
+    // Named Entity
+
+    /**
+     * Returns a list of general named entities in the provided text. For a list of supported entity types, check:
+     * https://aka.ms/taner For a list of enabled languages, check: https://aka.ms/talangs
+     *
+     * @param text the text to recognize entities for.
+     *
+     * @return A {@link Mono} containing the {@link RecognizeEntitiesResult named entity} of the text.
+     *
+     * @throws NullPointerException if {@code text} is {@code null}.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<RecognizeEntitiesResult> recognizeEntities(String text) {
+        try {
+            return recognizeEntitiesWithResponse(text, defaultLanguage).flatMap(FluxUtil::toMono);
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    /**
+     * Returns a list of general named entities in the provided text. For a list of supported entity types, check:
+     * https://aka.ms/taner For a list of enabled languages, check: https://aka.ms/talangs
+     *
+     * @param text the text to recognize entities for.
+     * @param language The 2 letter ISO 639-1 representation of language. If not set, uses "en" for English as
+     * default.
+     *
+     * @return A {@link Mono} containing a {@link Response} whose {@link Response#getValue() value} has the {@link
+     * RecognizeEntitiesResult named entity} of the text.
+     *
+     * @throws NullPointerException if {@code text} is {@code null}.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<RecognizeEntitiesResult>> recognizeEntitiesWithResponse(String text, String language) {
+        try {
+            return withContext(context -> recognizeEntitiesWithResponse(text, language, context));
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    Mono<Response<RecognizeEntitiesResult>> recognizeEntitiesWithResponse(String text, String language,
+        Context context) {
+        return recognizeBatchEntitiesWithResponse(
+            Arrays.asList(new TextDocumentInput(Integer.toString(0), text, language)), null, context)
```

Comment:
Use Collections.singletonList rather than Arrays.asList because there is only one item in it.
It's easier to use a constant "0" instead of creating an object and then doing toString on it.

You don't need to use flatMap. .map is sufficient because there is nothing async about this. Then you don't have to allocate and wrap the response with Mono.just